hw/ip/edn/data/edn.hjson

```diff
@@ -0,0 +1,276 @@
+                state machine command has been completed, at which point is safe to issue custom
+                commands via the software. command forwarding interface. A value of zero indicates
+                that all commands have completed, and the state machine is idle.
+                '''
+        }
+        { bits: "1",
+        name: "BOOT_INST_ACK",
+          desc: '''
+                This bit indicates that the EDN state machine has started the instantiation
+                process. If this bit is asserted it means the state machine has started in
+                boot-time request mode and that the the CSRNG has confirmed that this
+                EDN will be the next application to receive a seed from the entropy_src IP.
+                '''
+        }
+        { bits: "31",
+          name: "DIAG",
+          desc: "This this an internal generic diagnostic bit."
+        }
+      ]
+    },
+    { name: "SW_CMD_REQ",
+      desc: "EDN csrng app command request register",
+      swaccess: "wo",
+      hwaccess: "hro",
+      hwext: "true",
+      hwqe: "true",
+      regwen: "REGEN",
+      fields: [
+        { bits: "31:0",
+          name: "SW_CMD_REQ",
+          desc: '''
+                Writing this reqister with defined CSRNG commands will initiate any
+                possible CSRNG action. The application interface must wait for the
+                "ack" to return before issuing new commands. This interface is intended
+                to be controlled solely by software. If AUTO_REQ_MODE is set, this
+                register will have no effect on operation.
+                '''
+        }
+      ]
+    },
+    { name: "SW_CMD_STS",
+      desc: "EDN command status register",
+      swaccess: "rw",
+      hwaccess: "hwo",
+      fields: [
+        { bits: "0",
+          name: "CMD_RDY",
+          desc: '''
+                This bit indicates when the command interface is ready to accept commands.
+                '''
+        }
+        { bits: "1",
+          name: "CMD_ACK",
+          desc: '''
+                This bit is set when the command request has completed. This bit must
+                be cleared before a new request is made.
+                '''
+        }
+        { bits: "4",
+          name: "CMD_STS",
+          desc: '''
+                This one bit field is the status code returned with the application command ack.
+                0b0: Request completed successfully
+                0b1: Request completed with an error
+                '''
+        }
+      ]
+    },
+    { name: "RESEED_CMD",
+      desc: "EDN csrng reseed command register",
+      swaccess: "wo",
+      hwaccess: "hro",
+      hwext: "true",
+      hwqe: "true",
+      regwen: "REGEN",
+      fields: [
+        { bits: "31:0",
+          name: "RESEED_CMD",
+          desc: '''
+                Writing this register will fill a FIFO with up to 13 command words (32b words).
```

Comment:
Yes.  In the CTRL register, bit 1 is now CMD_FIFO_RST.   Setting this register clears all three FIFOs.  Change will come in an upcoming commit.

Reply:
Thanks for the reset bit. I still think we need to specify what happens if I write too many values to the (circular) FIFO.

Reply:
Oh good point.   I missed that.  I will add a comment saying that they just happily pop what's on HEAD.  What do you think?

Reply:
Sounds sensible to me: I think either end (or the middle?!) is fine, but it probably is good to specify.

Reply:
For this specific case, FW will create a FIFO hardware error if it tries to write a FIFO without space (this is documented in the interrupt section). Also, as a general ASIC rule that I have followed, you do not have to document all of the ways a chip will behave if you program it incorrectly. However, proper sequences should be provided to avoid confusion.

Reply:
I hate to be late to the party, on this thread but when we get into security functionality, we usually do have to specify all the error/failure modes @mwbranstad  :(

One of the benefits of doing that is that it behooves us to have a single/few error handling/recovery paths that we can take and generally steer the design into following those when Bad Stuff Happens (tm).

Reply:
@cdgori that makes sense, but I was talking about how FW can create a hardware error if designed incorrectly. If that is what you mean, then we can try to cover all cases, but given that FW can do about anything, just concerned about this possible challenge.

Reply:
The really big problem is that malicious actors who hijack the FW can really do *anything* so we have to try to prevent shooting-oneself-in-the-foot as much as we can from a "hardware API" standpoint.  Obviously there are cases where that's just not possible, so let's just do the best we can.